### ISSUE TITLE
Extra trailing line gets inserted to code blocks on saving

### ISSUE THREAD

**Issue by Akjir** [nextcloud/text#150]
**Describe the bug**
After i reopened a saved file, the formatting of the code block is wrong. See screenshots.

**To Reproduce**
1. create a new file
2. write Text
3. combine with code block
4. save and close
5. reopen

**Expected behavior**
The reopened file should have the same formatting like before closing.

**Screenshots**

File i created:

![1](https://user-images.githubusercontent.com/2734278/60916757-82cce580-a28f-11e9-91d7-b2d6426797d9.JPG)

Plain text:

![2](https://user-images.githubusercontent.com/2734278/60916780-92e4c500-a28f-11e9-98a0-1b1e76e0dd51.JPG)

After saving and reopening:

![3](https://user-images.githubusercontent.com/2734278/60916787-96784c00-a28f-11e9-9ab4-f5cac24a9db2.JPG)

**Client details:**
 - Windows 10 Pro 64bit
 - Firefox 67.0.4 64bit
 - Nextcloud 16.0.2
 - Nextcloud Text 1.0.0

Edit: It seems the code block uses the blank line.

**Comment by claell** [comment]
Might be related or have the same root cause as the plenty issues described in #593, so possibly a duplicate.

**Comment by herrdeh** [comment]
Still present with NC 20.0.7, markdown editor (by Robin Appelman) 2.3.3, qownnotes 21.1.3, [see here.](https://help.nextcloud.com/t/bug-markdown-code-block-empty-line-at-end-of-block/115224)

**Comment by juliushaertl** [comment]
Seems to happen only when using code blocks in lists, especially if the last line with the code block separator `\`\`\`` is indented:

```js
	test('failure with newline', () => {
		const text = `
### Headline

* Some list item

  \`\`\`
  nano  /var/www/html/config/config.php
  \`\`\``
		// markdown it already shows the issue
		// expect(markdownit.render(text)).toBe("")
		expect(markdownThroughEditor(markdownThroughEditor(text))).toBe(text)
	})
```

The generated HTML that markdown it generates already contains a newline:

```
    expect(received).toBe(expected) // Object.is equality

    Expected: ""
    Received: "<h3>Headline</h3>
    <ul>
    <li>
    <p>Some list item</p>
    <pre><code>nano  /var/www/html/config/config.php
    </code></pre>
    </li>
    </ul>
    "

      190 |   \`\`\``
      191 | 		// markdown it already shows the issue
    > 192 | 		expect(markdownit.render(text)).toBe("")
          | 		                                ^
      193 | 		expect(markdownThroughEditor(markdownThroughEditor(text))).toBe(text)
      194 | 	})
      195 |
```

Not sure yet how to address that best yet though.

**Comment by XueSheng-GIT** [comment]
> Seems to happen only when using code blocks in lists

Just creating a new file (md), adding a code block with text, saving/closing and re-opening does already trigger this issue (new line). I'm on NC24.0.1.

**Comment by the-s-a-m** [comment]
Issues still exist. Please resolve the new line issue in code blocks

**Comment by susnux** [comment]
> Just creating a new file (md), adding a code block with text, saving/closing and re-opening does already trigger this issue (new line). I'm on NC24.0.1.

For me this happens only in the editor, but the markdown source file does not contain the newline.
(on the master branch)

**Comment by max-nextcloud** [comment]
I think this is the same issue that was reported in #2344 which was fixed in #3321.
So closing this.
